elasticdl/python/worker/worker.py

```diff
@@ -468,14 +470,38 @@ def report_gradient_to_ps(self, grads):
         ):
             ps_id = self._var_to_ps[v.name]
             if ps_id not in ps_grads:
-                ps_grads[ps_id] = [(g, v.name)]
+                ps_grads[ps_id] = {v.name: g}
             else:
-                ps_grads[ps_id].append((g, v.name))
+                if v.name not in ps_grads[ps_id]:
+                    ps_grads[ps_id][v.name] = g
+                else:
+                    if isinstance(g, tf.IndexedSlices):
+                        ps_grads[ps_id][v.name] = merge_indexed_slices(
+                            ps_grads[ps_id][v.name], g
+                        )
+                    else:
+                        ps_grads[ps_id][v.name] += g
```

Comment:
Why  `ps_grads[ps_id][v.name] += g` not `ps_grads[ps_id][v.name] = g` ?

Reply:
Please review #1749 first.

```python
                if v.name not in ps_grads[ps_id]:
                    ps_grads[ps_id][v.name] = g
                else:
                    if isinstance(g, tf.IndexedSlices):
                        ps_grads[ps_id][v.name] = merge_indexed_slices(
                            ps_grads[ps_id][v.name], g
                        )
                    else:
                        ps_grads[ps_id][v.name] += g
```

If the gradient comes for the first time, we `ps_grads[ps_id][v.name] = g`.
If the gradient with a same name for the second time, we will merge the gradient to the existing gradient.

For `IndexedSlices` type gradient, we merge them;
For Dense type gradient, we add them

Reply:
Got it!